plugin/src/main/java/org/opensearch/ml/processor/ModelExecutor.java

```diff
@@ -0,0 +1,218 @@
+                        ModelTensor tensor = output.getMlModelTensors().get(0);
+                        // try getDataAsMap first
+                        Map<String, ?> tensorInDataAsMap = tensor.getDataAsMap();
+                        if (tensorInDataAsMap != null) {
+                            modelOutputValue = getModelOutputField(tensorInDataAsMap, originalModelOutputFieldName, ignoreMissing);
+                        }
+                        // if dataAsMap is empty try getData
+                        else {
+                            // pase data type
+                            modelOutputValue = parseGetDataInTensor(tensor);
+                        }
+                    } else {
+
+                        // for multiple tensors, initiate an array
+                        ArrayList tensorArray = new ArrayList<>();
+                        for (int i = 0; i < output.getMlModelTensors().size(); i++) {
+                            ModelTensor tensor = output.getMlModelTensors().get(i);
+
+                            // Adding null check for tensor
+                            if (tensor != null) {
+                                // Assuming getData() method may throw an exception
+                                // if the data is not available or is in an invalid state.
+                                try {
+                                    // try getDataAsMap first
+                                    Map<String, ?> tensorInDataAsMap = tensor.getDataAsMap();
+                                    if (tensorInDataAsMap != null) {
+
+                                        tensorArray
+                                            .add(getModelOutputField(tensorInDataAsMap, originalModelOutputFieldName, ignoreMissing));
+                                    }
+                                    // if dataAsMap is empty try getData
+                                    else {
+                                        tensorArray.add(parseGetDataInTensor(tensor));
+                                    }
+                                } catch (Exception e) {
+                                    // Handle the exception accordingly
+                                    throw new RuntimeException("Error accessing tensor data: " + e.getMessage());
+                                }
+                            }
+                        }
+                        modelOutputValue = tensorArray;
+                    }
+
+                } else {
+                    throw new RuntimeException("Output tensors are null or empty.");
+                }
+            } else {
+                throw new RuntimeException("Model outputs are null or empty.");
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("An unexpected error occurred: " + e.getMessage());
+        }
+        return modelOutputValue;
+    }
+
+    private static Object parseGetDataInTensor(ModelTensor tensor) {
```

Comment:
```
import java.util.Arrays;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

private static Object parseGetDataInTensor(ModelTensor tensor) {
    if (tensor.getDataType().isInteger()) {
        return IntStream.of(tensor.getData()).boxed().collect(Collectors.toList());
    } else if (tensor.getDataType().isFloating()) {
        return Arrays.stream(tensor.getData()).asDoubleStream().boxed().collect(Collectors.toList());
    } else if (tensor.getDataType().isString()) {
        return Arrays.stream(tensor.getData()).mapToObj(String::valueOf).collect(Collectors.toList());
    } else if (tensor.getDataType().isBoolean()) {
        return IntStream.of(tensor.getData()).mapToObj(num -> num != 0).collect(Collectors.toList());
    } else {
        throw new RuntimeException("unsupported data type in prediction data.");
    }
}
```

try this?

Reply:
Cannot resolve method 'asDoubleStream' in 'Stream'